doc/dev/adr/0020-large-message-chunking.md

```diff
@@ -0,0 +1,142 @@
+  - QoS settings are maintained across all chunks.
+- Receiving Process:
+  - The Chunking aware client receives messages and identifies chunked messages by the presence of chunk metadata.
+  - Chunks are stored in a temporary buffer, indexed by message ID and chunk index.
+  - When all chunks for a message ID are received, they are reassembled in order and message checksum verified (see Checksum Algorithm Options for MQTT Message Chunking).
+  - The reconstructed message is then processed as a single message by the application.
+- Receiving Failures:
+  - Message timeout interval ended before all chunks received.
+  - Reassembly buffer size limit reached before all chunks received.
+  - Calculated checksum does not match checksumm from chunk metadata.
+
+**Configuration settings:**
+- Enable/Disable
+- Overhead size
+- Reassembly buffer size limit
+
+### Implementation Considerations
+
+- **Error Handling:**
+  - Chunk timeout mechanisms (see Chunk Timeout Mechanism)
+  - Error propagation to application code
+- **Performance Optimization:**
+  - Concurrent chunk transmission
+  - Efficient memory usage during reassembly
+  - Maximum reassembly buffer size limits (configurable)
+
+### Benefits
+
+- **Property Preservation:** maintains topic, QoS, and other message properties consistently.
+- **Network Optimized:** allows efficient transmission of large payloads over constrained networks.
+
+### Compatibility
+
+- Non-chunking-aware clients will receive individual chunks as separate messages. Chunks reassembly could be implemented on the application side, given chunking implementation is known.
+
+## Appendix
+
+### Message Flow Diagram
+
+```mermaid
+sequenceDiagram
+    participant Sender as Sending Client
+    participant Broker as MQTT Broker
+    participant Receiver as Receiving Client
+
+    Note over Sender: Large message (>max size)
+    Sender->>Sender: Split into chunks
+    Note right of Sender: Calculate chunk size:<br/>MaxPacketSize - Overhead
+
+    loop For each chunk
+        Sender->>Broker: MQTT PUBLISH with __chunk metadata
+        Note over Broker: No special handling<br/>required by broker
+        Broker->>Receiver: Forward chunk
+        Note over Receiver: First chunk starts timeout countdown
+        Receiver->>Receiver: Store in buffer
+        Note over Receiver: Index by:<br/>messageId + chunkIndex
+    end
+
+    alt Success Path
+        Note over Receiver: All chunks received
+        Receiver->>Receiver: Verify checksum
+        Note right of Receiver: SHA-256 or<br/>custom algorithm
+        Receiver->>Receiver: Reassemble message
+        Note over Receiver: Process complete message
+    else Failure: Timeout
+        Note over Receiver: Message Expiry Interval exceeded
+        Receiver->>Receiver: Timeout occurred
+        Receiver->>Receiver: Cleanup buffers
+        Note over Receiver: Notify application:<br/>ChunkTimeoutError
```

Comment:
we will need some new rpc errors for communicating chunking issues to the invoker in the command response